src/main/java/duke/tasks/Task.java

```diff
@@ -0,0 +1,47 @@
+package duke.tasks;
+
+import java.util.HashSet;
+
+/** Abstract class representing a task */
+public abstract class Task {
+
+    private boolean done = false;
+    private String taskDetails;
+
+    public Task(String taskDetails) {
+        this.taskDetails = taskDetails;
+    }
+
+    public boolean markDone() {
```

Comment:
Perhaps you could add a JavaDoc comment here, so that others will know what the returned boolean value represents